packages/components/src/search-control/index.tsx

```diff
@@ -31,6 +32,14 @@ function SuffixItem( {
 		return <Icon icon={ search } />;
 	}
 
+	if ( onClose ) {
+		deprecated( 'onClose function prop', {
+			since: '6.7',
+			version: '',
+			plugin: 'wp.components.SearchControl',
+		} );
```

Comment:
- Let's phrase it this way so the component name is visible in the console warning. (Apparently the `plugin` string won't show unless [`version` is truthy](https://github.com/WordPress/gutenberg/blob/e86152bc8b45d5fe213ed688f145e82f73bb2fd4/packages/deprecated/src/index.js#L46-L48)).
- This isn't going to land in 6.7, so it'll be the next release.
- We can just omit the `version` property.

```suggestion
		deprecated( '`onClose` prop in wp.components.SearchControl', {
			since: '6.8',
		} );
```